oauth_provider/oauth2/validator.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 SYLEAM
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime, timedelta
+from openerp import http
+from openerp import fields
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from oauthlib.oauth2 import RequestValidator
+except ImportError:
+    _logger.debug('Cannot `import oauthlib`.')
+
+
+class OdooValidator(RequestValidator):
+    """
+    OAuth2 validator to be used in Odoo
+    """
+    def authenticate_client(self, request, *args, **kwargs):
+        """
+        Authenticate the client
+        """
+        raise NotImplementedError
+        # return False
+
+    def authenticate_client_id(self, client_id, request, *args, **kwargs):
+        """
+        Ensure client_id belong to a non-confidential client
+        """
+        request.client = http.request.env['oauth.provider.client'].search([
+            ('identifier', '=', client_id),
+        ])
+        request.odoo_user = http.request.env['oauth.provider.client'].env.user
```

Comment:
Why the `env` and `user` from `oauth.provider.client`? It's the same `env` and `user` isn't it, or is there some fancy context switching that I am missing?